src/main/java/org/springframework/cloud/deployer/spi/scheduler/cloudfoundry/CloudFoundryAppScheduler.java

```diff
@@ -220,8 +257,14 @@ private void scheduleTask(String appName, String scheduleName,
 	private String stageTask(ScheduleRequest scheduleRequest) {
 		logger.debug(String.format("Staging Task: ",
 				scheduleRequest.getDefinition().getName()));
+		Map<String, String> properties = new HashMap<>(scheduleRequest.getDefinition().getProperties());
+		properties.put(SCHEDULER_TASK_DEF_NAME_KEY,
+				scheduleRequest.getDefinition().getName());
+		this.scheduleTaskMap.put(scheduleRequest.getScheduleName(), scheduleRequest.getDefinition().getName());
```

Comment:
I think the key and value are always going to be same here.

Reply:
Not in this case.  You can test this by building this PR and then building SCDF using this PR.